Unable to calculate the type of a varargs parameter.

I have received an exception when trying to symbol solve the following method:

```
public void execute(final ComponentBase _this) {
        String regexText = null;
        String functionName = "getRegexValues";
	final ObjectContext scemAdapter = _this.getChildByAlias("scem_adapter");
        final Integer outputIdx = scemAdapter.getListString("NWT_RESULTADO_TEXT").indexOf("asyn");
        if (outputIdx >= 0) {
            String outpuText = scemAdapter.getListString("NWT_RESULTADO_TEXT_VALUE").get(outputIdx);
	}
	_this.getParent().set("NWT_RESULTADO_INERCIA", _this.getRoot().executeMethod(functionName,regexText,outpuText,1));
}
```

```
java.lang.RuntimeException: Unable to calculate the type of a parameter of a method call. Method call: _this.getParent().set("NWT_RESULTADO_IEC_ETA_100", _this.getRoot().executeMethod(functionName, regexText, outpuText, 5)), Parameter: _this.getRoot().executeMethod(functionName, regexText, outpuText, 5)
```

The class referenced by `_this` is `ComponentBase.class`, that extends `ObjectContext.class`. Both classes can be found inside a jar file that I added to my `CombinedTypeSolver` via `JarTypeSolver`. I have also added a `new ReflectionTypeSolver()` to the configuration. The method `executeMethod` can be found in `ObjectContext.class` (retrieved by the `getRoot()` method) has the following signature:

```
public Object executeMethod(String methodName, Object... values) {
        //...
}
```

The same exception was thrown on other java files whenever `executeMethod` was called. Since both dependencies were correctly added to the `CombinedTypeSolver`, I wonder:

- Is the symbol solver capable of tracing the reference of a method that takes varargs as a parameter?
- Are there any other reasons of why this might be occurring? Am I missing something?

@jlerbsc sure thing! Here it is:

```
java.lang.RuntimeException: Unable to calculate the type of a parameter of a method call. Method call: _this.getParent().set("NWT_RESULTADO_IEC_ETA_100", _this.getRoot().executeMethod(functionName, regexText, outpuText, 5)), Parameter: _this.getRoot().executeMethod(functionName, regexText, outpuText, 5)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solveArguments(JavaParserFacade.java:225)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solve(JavaParserFacade.java:240)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solve(JavaParserFacade.java:125)
	at com.github.javaparser.symbolsolver.JavaSymbolSolver.resolveDeclaration(JavaSymbolSolver.java:140)
	at com.github.javaparser.ast.expr.MethodCallExpr.resolve(MethodCallExpr.java:313)
	at net.weg.parserpoc.parser.RuleStatisticsDTO.processMethod(RuleStatisticsDTO.java:27)
	at net.weg.parserpoc.parser.JavaParser.lambda$insertMethodsToDTO$0(JavaParser.java:29)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at net.weg.parserpoc.parser.JavaParser.insertMethodsToDTO(JavaParser.java:29)
	at net.weg.parserpoc.parser.JavaParser.findMethodCallsOn(JavaParser.java:24)
	at net.weg.parserpoc.Programs.lambda$findMethodCalls$0(Programs.java:41)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at net.weg.parserpoc.Programs.findMethodCalls(Programs.java:41)
	at net.weg.parserpoc.Programs.init(Programs.java:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.springframework.test.context.junit4.statements.RunBeforeTestExecutionCallbacks.evaluate(RunBeforeTestExecutionCallbacks.java:74)
	at org.springframework.test.context.junit4.statements.RunAfterTestExecutionCallbacks.evaluate(RunAfterTestExecutionCallbacks.java:84)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:251)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.RuntimeException: Method 'executeMethod' cannot be resolved in context _this.getRoot().executeMethod(functionName, regexText, outpuText, 1) (line: 144) MethodCallExprContext{wrapped=_this.getRoot().executeMethod(functionName, regexText, outpuText, 1)}. Parameter types: [ReferenceType{java.lang.String, typeParametersMap=TypeParametersMap{nameToValue={}}}, ReferenceType{java.lang.String, typeParametersMap=TypeParametersMap{nameToValue={}}}, ReferenceType{java.lang.String, typeParametersMap=TypeParametersMap{nameToValue={}}}, PrimitiveTypeUsage{name='int'}]
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solveMethodAsUsage(JavaParserFacade.java:587)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:267)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:44)
	at com.github.javaparser.ast.expr.MethodCallExpr.accept(MethodCallExpr.java:115)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getTypeConcrete(JavaParserFacade.java:448)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:310)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solveArguments(JavaParserFacade.java:221)
	... 43 more
```

--- a/javaparser-core/src/main/java/com/github/javaparser/resolution/logic/MethodResolutionLogic.java
+++ b/javaparser-core/src/main/java/com/github/javaparser/resolution/logic/MethodResolutionLogic.java
@@ -798,30 +798,8 @@ public class MethodResolutionLogic {
         return paramType.isReferenceType() && paramType.asReferenceType().getQualifiedName().equals("java.lang.Object");
     }
 
-    private static boolean isMoreSpecific(MethodUsage methodA, MethodUsage methodB) {
-        boolean oneMoreSpecificFound = false;
-        for (int i = 0; i < methodA.getNoParams(); i++) {
-            ResolvedType tdA = methodA.getParamType(i);
-            ResolvedType tdB = methodB.getParamType(i);
-
-            boolean aIsAssignableByB = tdA.isAssignableBy(tdB);
-            boolean bIsAssignableByA = tdB.isAssignableBy(tdA);
-
-            // A is more specific
-            if (bIsAssignableByA && !aIsAssignableByB) {
-                oneMoreSpecificFound = true;
-            }
-            // B is more specific
-            if (aIsAssignableByB && !bIsAssignableByA) {
-                return false;
-            }
-
-            // If B is vararg and A is not, A is more specific
-            if (tdB.isArray() && tdB.asArrayType().getComponentType().isAssignableBy(tdA)) {
-                oneMoreSpecificFound = true;
-            }
-        }
-        return oneMoreSpecificFound;
+    private static boolean isMoreSpecific(MethodUsage methodA, MethodUsage methodB, List<ResolvedType> argumentTypes) {
+    	return isMoreSpecific(methodA.getDeclaration(), methodB.getDeclaration(), argumentTypes);
     }
 
     public static Optional<MethodUsage> findMostApplicableUsage(List<MethodUsage> methods, String name, List<ResolvedType> argumentsTypes, TypeSolver typeSolver) {
@@ -836,9 +814,9 @@ public class MethodResolutionLogic {
         MethodUsage winningCandidate = applicableMethods.get(0);
         for (int i = 1; i < applicableMethods.size(); i++) {
                 MethodUsage other = applicableMethods.get(i);
-                if (isMoreSpecific(winningCandidate, other)) {
+                if (isMoreSpecific(winningCandidate, other, argumentsTypes)) {
                     // nothing to do
-                } else if (isMoreSpecific(other, winningCandidate)) {
+                } else if (isMoreSpecific(other, winningCandidate, argumentsTypes)) {
                     winningCandidate = other;
                 } else {
                     if (winningCandidate.declaringType().getQualifiedName().equals(other.declaringType().getQualifiedName())) {

